spec/lib/manageiq/api/common/request_spec.rb

```diff
@@ -0,0 +1,47 @@
+describe ManageIQ::API::Common::Request do
+  let(:request_good) do
+    headers = ActionDispatch::Http::Headers.from_hash(:blah           => 'blah',
+                                                      'x-rh-identity' => 'abc')
+    ActionDispatch::Request.new(headers)
+  end
+  let(:request_bad) { {:blah => 'blah' } }
+
+  context "user" do
+    it "returns a lazy loaded user instance" do
+      described_class.with_request(request_good) do |instance|
+        expect(instance.user).to be_a(ManageIQ::API::Common::User)
+      end
+    end
+  end
+
+  context "set current headers" do
```

Comment:
I think you also need a test where the with_request is called with a Hash to a) test that code path and b) show that we can reconstruct this object in non-controller environments.